internal/datasource/config/config_writer_test.go

```diff
@@ -131,6 +131,73 @@ func TestWriteConfig(t *testing.T) {
 	helpers.RemoveFileWithErrorCheck(t, nginxConf.Name())
 }
 
+func TestDeleteFile(t *testing.T) {
+	tempDir := t.TempDir()
+	_, instanceID := helpers.CreateTestIDs(t)
+
+	agentconfig := types.GetAgentConfig()
+
+	instanceIDDir := path.Join(tempDir, instanceID.String())
+	helpers.CreateDirWithErrorCheck(t, instanceIDDir)
+	defer helpers.RemoveFileWithErrorCheck(t, instanceIDDir)
+
+	testConf := helpers.CreateFileWithErrorCheck(t, tempDir, "test.conf")
+	nginxConf := helpers.CreateFileWithErrorCheck(t, tempDir, "nginx.conf")
+	metricsConf := helpers.CreateFileWithErrorCheck(t, tempDir, "metrics.conf")
+
+	fileCacheContent, getCacheErr := protos.GetFileCache(nginxConf, testConf, metricsConf)
+	require.NoError(t, getCacheErr)
+
+	currentFileCache, getCacheErr := protos.GetFileCache(nginxConf, metricsConf)
+	require.NoError(t, getCacheErr)
+
+	tests := []struct {
+		name             string
+		fileCache        CacheContent
+		currentFileCache CacheContent
+		fileDeleted      bool
+	}{
+		{
+			name:             "file doesn't need deleting",
+			fileCache:        fileCacheContent,
+			currentFileCache: fileCacheContent,
+			fileDeleted:      false,
+		},
+		{
+			name:             "file needs deleting",
+			fileCache:        fileCacheContent,
+			currentFileCache: currentFileCache,
+			fileDeleted:      true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cacheFile := helpers.CreateFileWithErrorCheck(t, instanceIDDir, "cache.json")
+			cachePath := cacheFile.Name()
+			fileCache := NewFileCache(instanceID.String())
+			fileCache.SetCachePath(cachePath)
+			err := fileCache.UpdateFileCache(test.fileCache)
+			require.NoError(t, err)
+			slog.Info("", "", &agentconfig)
+			configWriter, err := NewConfigWriter(agentconfig, fileCache)
+			require.NoError(t, err)
+
+			configWriter.removeFiles(test.currentFileCache, test.fileCache)
+
+			if test.fileDeleted {
+				assert.NoFileExists(t, testConf.Name())
+			} else {
+				assert.FileExists(t, testConf.Name())
+			}
+			helpers.RemoveFileWithErrorCheck(t, cacheFile.Name())
+		})
+	}
+
+	helpers.RemoveFileWithErrorCheck(t, metricsConf.Name())
```

Comment:
This functions might not get called if the unit test fails. Its best to call `defer helpers.RemoveFileWithErrorCheck` straight away after calling `helpers.CreateFileWithErrorCheck` to insure that the file gets deleted at the end of the test.